package/MDAnalysis/coordinates/H5MD.py

```diff
@@ -726,6 +772,670 @@ def __setstate__(self, state):
+    file can be written with a contiguous layout by setting ``chunks=False``,
+    however this must be accompanied by setting ``n_frames`` equal to the
+    number of frames being written, as HDF5 must know how much space to
+    allocate on disk when creating the dataset.
+
+    .. _`H5PY compression options`: https://docs.h5py.org/en/stable/high/dataset.html#filter-pipeline
+
+
+    .. versionadded:: 2.0.0
+
+    """
+
+    format = 'H5MD'
+    multiframe = True
+    #: These variables are not written from :attr:`Timestep.data`
+    #: dictionary to the observables group in the H5MD file
+    data_blacklist = ['step', 'time', 'dt']
+
+    #: currently written version of the file format
+    H5MD_VERSION = (1, 1)
+
+    # This dictionary is used to translate MDAnalysis units to H5MD units.
+    # (https://nongnu.org/h5md/modules/units.html)
+    _unit_translation_dict = {
+        'time': {
+            'ps': 'ps',
+            'fs': 'fs',
+            'ns': 'ns',
+            'second': 's',
+            'sec': 's',
+            's': 's',
+            'AKMA': 'AKMA'},
+        'length': {
+            'Angstrom': 'Angstrom',
+            'angstrom': 'Angstrom',
+            'A': 'Angstrom',
+            'nm': 'nm',
+            'pm': 'pm',
+            'fm': 'fm'},
+        'velocity': {
+            'Angstrom/ps': 'Angstrom ps-1',
+            'A/ps': 'Angstrom ps-1',
+            'Angstrom/fs': 'Angstrom fs-1',
+            'A/fs': 'Angstrom fs-1',
+            'Angstrom/AKMA': 'Angstrom AKMA-1',
+            'A/AKMA': 'Angstrom AKMA-1',
+            'nm/ps': 'nm ps-1',
+            'nm/ns': 'nm ns-1',
+            'pm/ps': 'pm ps-1',
+            'm/s': 'm s-1'},
+        'force':  {
+            'kJ/(mol*Angstrom)': 'kJ mol-1 Angstrom-1',
+            'kJ/(mol*nm)': 'kJ mol-1 nm-1',
+            'Newton': 'Newton',
+            'N': 'N',
+            'J/m': 'J m-1',
+            'kcal/(mol*Angstrom)': 'kcal mol-1 Angstrom-1',
+            'kcal/(mol*A)': 'kcal mol-1 Angstrom-1'}}
+
+    @due.dcite(Doi("10.25080/majora-1b6fd038-005"),
+               description="MDAnalysis trajectory reader/writer of the H5MD"
+               "format", path=__name__)
+    @due.dcite(Doi("10.1016/j.cpc.2014.01.018"),
+               description="Specifications of the H5MD standard",
+               path=__name__, version='1.1')
+    def __init__(self, filename, n_atoms, n_frames=None, driver=None,
+                 comm=None, convert_units=True, chunks=None, compression=None,
+                 compression_opts=None, positions=True, velocities=True,
+                 forces=True, timeunit=None, lengthunit=None,
+                 velocityunit=None, forceunit=None, author='N/A',
+                 author_email=None, creator='MDAnalysis',
+                 creator_version=mda_version, **kwargs):
+
+        if not HAS_H5PY:
+            raise RuntimeError("H5MDWriter: Please install h5py")
```

Comment:
We have the machinery to mock the inability to import of a module. Can we do this to test this?

Reply:
Check out PR #3294 where @IAlibay mocks out netcdf.